ApplicationLibCode/Commands/OperationsUsingObjReferences/RicCopyReferencesToClipboardFeature.cpp

```diff
@@ -124,63 +125,37 @@ bool RicCopyReferencesToClipboardFeature::isAnyCopyableObjectSelected()
 //--------------------------------------------------------------------------------------------------
 bool RicCopyReferencesToClipboardFeature::isCopyOfObjectSupported( caf::PdmObject* pdmObject )
 {
+    // Copy support based on direct dynamic cast
+    if ( dynamic_cast<RimGeoMechView*>( pdmObject ) ) return true;
+    if ( dynamic_cast<RimEclipseView*>( pdmObject ) ) return true;
+    if ( dynamic_cast<RimEclipseCase*>( pdmObject ) ) return true;
+    if ( dynamic_cast<RimExtrudedCurveIntersection*>( pdmObject ) ) return true;
+    if ( dynamic_cast<RimBoxIntersection*>( pdmObject ) ) return true;
+    if ( dynamic_cast<RimSummaryPlot*>( pdmObject ) ) return true;
+    if ( dynamic_cast<RimFractureTemplate*>( pdmObject ) ) return true;
+    if ( dynamic_cast<RimEnsembleCurveSet*>( pdmObject ) ) return true;
+    if ( dynamic_cast<RimGridCrossPlotDataSet*>( pdmObject ) ) return true;
+    if ( dynamic_cast<RimModeledWellPath*>( pdmObject ) ) return true;
+    if ( dynamic_cast<RimSummaryMultiPlot*>( pdmObject ) ) return true;
+
+    // Copy support based combined logic
```

Comment:
Maybe the objects themselves should determine if they can be copied? E.g. pdmObject->isCopyable()?